### ISSUE TITLE
How to manipulate BottomNavigationBarItem gesture size?

### ISSUE THREAD

**Issue by Stainie** [flutter/flutter#23004]
## Steps to Reproduce

![screenshot_2018-10-12-13-41-37](https://user-images.githubusercontent.com/15278052/46868761-b32f4280-ce29-11e8-9266-852d49f72c38.png)

```dart
import 'dart:async';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:flutter/material.dart';

import '../pages/login.dart';
import '../pages/change_password.dart';
import '../pages/connect_retailer.dart' as retailer;
import '../pages/new_session.dart' as session;
import '../pages/my_products.dart' as products;
import '../pages/stats_insights.dart' as stats;
import '../pages/balance.dart' as balance;

import '../cache/domain_cache.dart';

import '../helping_scripts/click_handler.dart';

class MainMenu extends StatefulWidget {
  final int _routeTo;
  MainMenu([this._routeTo = -1]);

  @override
  _MainMenuState createState() => new _MainMenuState(_routeTo);
}

class _MainMenuState extends State<MainMenu>
    with SingleTickerProviderStateMixin {
  TabController tabController;
  String titleText = "My Sessions";
  List<String> titles = [
    "My Sessions",
    "Stats & Insights",
    "My Products",
    "My Wallet"
  ];

  final int _routeTo;
  ClickHandler handler = new ClickHandler();

  _MainMenuState(this._routeTo);

  @override
  void initState() {
    super.initState();

    tabController = new TabController(vsync: this, length: 4);
    tabController.addListener(_handleTitle);

    if (_routeTo != -1) {
      if (_routeTo == 4) {
        Navigator.push(
            context,
            MaterialPageRoute(
                builder: (context) => retailer.ConnectRetailerScreen()));
      } else {
        routeToPage(_routeTo);
      }
    }
  }

  @override
  void dispose() {
    tabController.dispose();
    super.dispose();
  }

  void _handleTitle() {
    setState(() {
      titleText = titles[tabController.index];
    });
  }

  Future<bool> saveUnlogged() async {
    SharedPreferences prefs = await SharedPreferences.getInstance();

    prefs.setBool("logged", false);
    prefs.setString("uuid", "");
    prefs.setString("token", "");

    Navigator.push(
        context, MaterialPageRoute(builder: (context) => LoginPage()));

    return true;
  }

  void routeToPage(int pageTo) {
    tabController.animateTo(pageTo);
    //Navigator.of(context).pop();

    setState(() {
      switch (pageTo) {
        case 0:
          setTitleText("My Sessions");
          break;
        case 1:
          setTitleText("Stats & Insights");
          break;
        case 2:
          setTitleText("My Products");
          break;
        case 3:
          setTitleText("My Wallet");
          break;
        default:
      }
    });
  }

  void setTitleText(String text) {
    setState(() {
      this.titleText = text;
    });
  }

  final List<Entry> data = <Entry>[
    Entry('My Symptoms', getSymptoms()),
    Entry('5 Session Documentation', getDates()),
  ];

  static List<Entry> getSymptoms() {
    List<Entry> entries = new List<Entry>();

    if (DomainCache.mySymptoms != null) {
      for (var i = 0; i < DomainCache.mySymptoms.length; i++) {
        Entry entry = new Entry(DomainCache.mySymptoms[i]);

        entries.add(entry);
      }
    }
    return entries;
  }

  static List<Entry> getDates() {
    List<Entry> entries = new List<Entry>();

    if (DomainCache.myLastFive != null) {
      for (var i = 0; i < DomainCache.myLastFive.length; i++) {
        Entry entry = new Entry(DomainCache.myLastFive[i]);

        entries.add(entry);
      }
    }
    return entries;
  }

  void routeToConnectRetailer() {
    Navigator.push(
        context,
        MaterialPageRoute(
            builder: (context) => retailer.ConnectRetailerScreen()));
  }

  void routeToChangePassword() {
    Navigator.push(
        context, MaterialPageRoute(builder: (context) => ChangePasswordPage()));
  }

  @override
  Widget build(BuildContext context) {
    final connectToRetailer = Padding(
      padding: EdgeInsets.symmetric(vertical: 16.0),
      child: Material(
        borderRadius: BorderRadius.circular(30.0),
        shadowColor: Colors.lightGreenAccent.shade100,
        child: MaterialButton(
          minWidth: 100.0,
          height: 42.0,
          onPressed: () => routeToConnectRetailer(),
          color: const Color(0xff00c9d2),
          child: Text(
            'Connect To Retailer',
            style: TextStyle(
              color: Colors.white,
              fontWeight: FontWeight.w700,
              fontFamily: "Roboto",
              fontStyle: FontStyle.normal,
            ),
          ),
        ),
      ),
    );

    final hamburgerMenu = Drawer(
      child: ListView(
        children: <Widget>[
          UserAccountsDrawerHeader(
            accountName: Text(
              "Uuid: ${DomainCache.myUser != null ? DomainCache.myUser.uuid : "User"}",
              style: TextStyle(
                color: Colors.white,
                fontWeight: FontWeight.w700,
                fontFamily: "Roboto",
                fontStyle: FontStyle.italic,
              ),
            ),
            accountEmail: Text("Active since: ${DomainCache.myUser.active}",
                style: TextStyle(
                    color: Colors.grey[400],
                    fontWeight: FontWeight.w700,
                    fontFamily: "Roboto",
                    fontStyle: FontStyle.normal,
                    fontSize: 16.0)),
            currentAccountPicture: GestureDetector(
              //onTap: () => routeToChangePassword(),
              child: CircleAvatar(
                  backgroundImage: AssetImage("assets/avatar2.png")),
            ),
            decoration: BoxDecoration(
                image: DecorationImage(
                    fit: BoxFit.fill,
                    image: AssetImage("assets/blue_bar.jpg"))),
          ),
          Container(
              height: 200.0,
              child: ListView.builder(
                  itemBuilder: (BuildContext context, int index) =>
                      EntryItem(data[index]),
                  itemCount: data.length)),
          // ListTile(
          //   title: Text(
          //     "New Session",
          //     style:
          //         TextStyle(color: Colors.black, fontWeight: FontWeight.bold),
          //   ),
          //   trailing: Icon(Icons.question_answer),
          //   onTap: () => routeFromDrawer(0),
          // ),
          // ListTile(
          //     title: Text(
          //       "Stats & Insights",
          //       style:
          //           TextStyle(color: Colors.black, fontWeight: FontWeight.bold),
          //     ),
          //     trailing: Icon(Icons.pie_chart),
          //     onTap: () => routeFromDrawer(1)),
          // ListTile(
          //     title: Text(
          //       "Connect to Retailer",
          //       style:
          //           TextStyle(color: Colors.black, fontWeight: FontWeight.bold),
          //     ),
          //     trailing: Icon(Icons.cast_connected),
          //     onTap: () => routeFromDrawer(2)),
          // ListTile(
          //     title: Text(
          //       "Products",
          //       style:
          //           TextStyle(color: Colors.black, fontWeight: FontWeight.bold),
          //     ),
          //     trailing: Icon(Icons.fastfood),
          //     onTap: () => routeFromDrawer(3)),
          // ListTile(
          //     title: Text(
          //       "Balance",
          //       style:
          //           TextStyle(color: Colors.black, fontWeight: FontWeight.bold),
          //     ),
          //     trailing: Icon(Icons.attach_money),
          //     onTap: () => routeFromDrawer(4)),
          ListTile(
              title: Text(
                "Version Number: 0.21",
                style: TextStyle(
                  color: Colors.red[400],
                  fontWeight: FontWeight.w700,
                  fontFamily: "Roboto",
                  fontStyle: FontStyle.normal,
                ),
              ),
              trailing: Icon(Icons.done)),
          ListTile(
              title: Text(
                "Log out",
                style: TextStyle(
                  color: Colors.grey[400],
                  fontWeight: FontWeight.w700,
                  fontFamily: "Roboto",
                  fontStyle: FontStyle.normal,
                ),
              ),
              trailing: Icon(Icons.offline_bolt),
              onTap: () => handler.clickActionOnce(saveUnlogged)),
          Divider(),
          ListTile(
            title: Text(
              "Close",
              style: TextStyle(
                color: Colors.grey[400],
                fontWeight: FontWeight.w700,
                fontFamily: "Roboto",
                fontStyle: FontStyle.normal,
              ),
            ),
            trailing: Icon(Icons.cancel),
            onTap: () => Navigator.of(context).pop(),
          ),
          connectToRetailer
        ],
      ),
    );

    return WillPopScope(
        onWillPop: () async => false,
        child: Scaffold(
            appBar: new AppBar(
              elevation: 0.0,
              centerTitle: true,
              iconTheme: new IconThemeData(color: Colors.grey),
              title: Text(titleText,
                  textAlign: TextAlign.center,
                  style: const TextStyle(
                      color: Colors.grey,
                      fontWeight: FontWeight.w400,
                      fontFamily: "Poppins",
                      fontStyle: FontStyle.normal,
                      fontSize: 16.0)),
              backgroundColor: Colors.white,
            ),
            drawer: hamburgerMenu,
            bottomNavigationBar: BottomNavigationBar(
                  type: BottomNavigationBarType.fixed,
                  fixedColor: const Color(0xff00c9d2),
                  onTap: routeToPage,
                  currentIndex: tabController.index,
                  items: [
                    BottomNavigationBarItem(
                        icon: Image.asset("assets/menu_home2.png"),
                        activeIcon: Image.asset("assets/menu_home_blue2.png"),
                        title: Text(
                          "Home",
                          style: TextStyle(
                              fontWeight: FontWeight.w400,
                              fontFamily: "Avenir",
                              fontStyle: FontStyle.normal,
                              fontSize: 10.0),
                        )),
                    BottomNavigationBarItem(
                        icon: Image.asset("assets/menu_stats2.png"),
                        activeIcon: Image.asset("assets/menu_stats_blue2.png"),
                        title: Text("Stats",
                            style: TextStyle(
                                fontWeight: FontWeight.w400,
                                fontFamily: "Avenir",
                                fontStyle: FontStyle.normal,
                                fontSize: 10.0))),
                    BottomNavigationBarItem(
                        icon: Image.asset("assets/menu_products2.png"),
                        activeIcon:
                            Image.asset("assets/menu_products_blue2.png"),
                        title: Text("Products",
                            style: TextStyle(
                                fontWeight: FontWeight.w400,
                                fontFamily: "Avenir",
                                fontStyle: FontStyle.normal,
                                fontSize: 10.0))),
                    BottomNavigationBarItem(
                        icon: Image.asset("assets/menu_wallet2.png"),
                        activeIcon: Image.asset("assets/menu_wallet_blue2.png"),
                        title: Text("Wallet",
                            style: TextStyle(
                                fontWeight: FontWeight.w400,
                                fontFamily: "Avenir",
                                fontStyle: FontStyle.normal,
                                fontSize: 10.0))),
                  ],
                ),
            body: TabBarView(
              controller: tabController,
              children: <Widget>[
                session.NewSessionScreen(),
                stats.StatsScreen(),
                products.MyProductsScreen(),
                balance.BalanceScreen()
              ],
            )));
  }
}

class Entry {
  Entry(this.title, [this.children = const <Entry>[]]);

  final String title;
  final List<Entry> children;
}

class EntryItem extends StatelessWidget {
  const EntryItem(this.entry);
  final Entry entry;

  Widget _buildTiles(Entry root) {
    if (root.children.isEmpty) return ListTile(title: Text(root.title));
    return ExpansionTile(
      key: PageStorageKey<Entry>(root),
      title: Text(
        root.title,
        style: TextStyle(
          color: Colors.grey[400],
          fontWeight: FontWeight.w500,
          fontFamily: "Roboto",
          fontStyle: FontStyle.normal,
        ),
      ),
      children: root.children.map(_buildTiles).toList(),
    );
  }

  @override
  Widget build(BuildContext context) {
    return _buildTiles(entry);
  }
}
```

```
[√] Flutter (Channel beta, v0.9.4, on Microsoft Windows [Version 10.0.17134.345], locale sr-Latn-RS)
    • Flutter version 0.9.4 at C:\Users\User\Documents\Flutter\flutter
    • Framework revision f37c235c32 (2 weeks ago), 2018-09-25 17:45:40 -0400
    • Engine revision 74625aed32
    • Dart version 2.1.0-dev.5.0.flutter-a2eb050044

[!] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at C:\Users\User\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 27.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[√] Android Studio (version 3.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[!] VS Code, 64-bit edition (version 1.28.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected devices (1 available)
    • ALE L21 • QLF7N15C16002830 • android-arm64 • Android 6.0 (API 23)
```

**Comment by jslavitz** [comment]
@Stainie  Are you talking about the vertical size of the navigation bar or something else?

**Comment by Stainie** [comment]
@jslavitz yes, exactly.

**Comment by jslavitz** [comment]
@Stainie After looking into this, it seems like it's not a bug, it's really just the intended functionality. Right now, the bottom navigation bar size is set to 56 pixels. We will look into allowing this value to be user editable, but in the mean time, you can edit the value of kBottomNavigationBarHeight in constants.dart.

**Comment by Stainie** [comment]
@jslavitz Thank you, but unfortunately, that does not solve the problem. My problem is in the space between the grey area (body of the screen), and clear white navbar area. You can see in the screenshot that the gesture detector behind the icon of the navbar is too big (and cannot be manually changed) and is the reason of additionally created space. I hope that explanation helps.

**Comment by darshankawar** [comment]
Hi @Stainie,
Can you confirm if this is still an issue or has this been resolved on latest stable or master version ?
Thanks.
